lib/utils.js

```diff
@@ -125,8 +168,9 @@ function Utils(config) {
      * @returns {string} id - the feature's identifier
      */
     utils.idFromRecord = function(record) {
-        var id = record.id.S || record.id;
-        var bits = id.split('!');
+        var index = record.index.S || record.index;
+        var bits = index.split('!');
+        bits.shift();
         bits.shift();
         return bits.join('!');
     };
```

Comment:
Given how much simplification has happened to the routines in `utils.js`, it might not make sense to make a separate module for them anymore. Are they even used in more than one place?